src/client/QXmppClient.cpp

```diff
@@ -320,6 +324,28 @@ bool QXmppClient::isConnected() const
     return d->stream->isConnected();
 }
 
+/// Returns true if the current client state is "active", false if it is
+/// "inactive". See XEP-0352 for details.
+
+bool QXmppClient::isActive() const
+{
+    return d->isActive;
+}
+
+/// Sets the client state as described in to XEP-0352
+
+void QXmppClient::setActive(bool active)
+{
+    if (active != d->isActive && d->stream->isClientStateIndicationEnabled()) {
+        d->isActive = active;
+        QString packet("<%1 xmlns='%2'/>");
```

Comment:
This is probably a microoptimisation given that network is involved, but `ns_csi` is statically known, so maybe it's worth pasting that into `packet` straight away?